src/librustc_trans/mir/block.rs

```diff
@@ -137,6 +137,8 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
                     this.set_debug_loc(&ret_bcx, terminator.source_info);
                     this.store_return(&ret_bcx, ret_dest, &fn_ty.ret, invokeret);
                 }
+
+                None
```

Comment:
This *should probably* be `invokeret`.

Reply:
That would be bad, since `InvokeInst` is unrelated to `CallInst` and does not have a `notail` attribute.

Reply:
They're both `CallSite` though?

Reply:
I don't think that has a `setTailCallKind` method.

Reply:
Oh, my bad, I just checked again and that is indeed that case. That's a big troubling, because an `invoke` can become a `call` through optimizations :/.

Reply:
We should probably panic if we need to add notail on invokes then. I'll do that to check that we don't emit any invokes here.

Reply:
It seems like we do emit invokes here.

Reply:
I guess the attribute should require non-Rust ABI which implies `nounwind`?

Reply:
Isn't `invoke` only used when you need cleanup? Normal `call` instructions can also unwind and should be used here

Reply:
Right, unwinding + cleanup.